egg/zoo/compo_vs_generalization/curriculum_games.py

```diff
@@ -0,0 +1,85 @@
+import torch
+from abc import abstracmethod
+from egg.core.reinforce_wrappers import SenderReceiverRnnReinforce
+from egg.zoo.compo_vs_generalization.data import mask_attributes
+from egg.zoo.compo_vs_generalization.losses import MaskedLoss
+
+class CurriculumGameWrapper(nn.Module):
+    """
+    Abstract game wrapper for games that implement a curriculum.
+    Child classes should have a curriculum_level attribute
+    as well as implement an update_curriculum_level method.
+    """
+    def __init__(self, game: SenderReceiverRnnReinforce):
+        self.game = game
+
+    def forward(self, sender_input, labels, receiver_input=None, aux_input=None):
+        return self.game(sender_input, labels, receiver_input=None, aux_input=None)
+
+    @abstractmethod
+    def update_curriculum_level(self):
+        """
+        This method is called by the trainer
+        and should schedule the game's curriculum
+        """
+        pass
+
+
+
+class GraduallyRevealAttributes(CurriculumGameWrapper):
+    """
+    In this game wrapper, all attributes are masked except the first {curriculum level} ones.
+    At first, only the first (leftmost) attribute is visible, then at each curriculum update,
+    the next attribute is revealed, until all attributes are visible.
+    """
+    def __init__(self,
+            game: SenderReceiverRnnReinforce,
+            n_attributes: int,
+            n_values: int,
+            mode: str
+            initial_n_unmasked_idxs: int=1
+        ):
+        valid_modes = ['left_to_right', 'random']
+        if mode not in valid_modes:
+            raise ValueError(f"Invalid mode {mode}. mode should be"
+                              "either 'left_to_right' or 'random'")
```

Comment:
Protip: you can do something like 

```python
f"mode should be in {valid_modes}"
```

this way you won't have to update the error message every time you add a new mode